src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -342,6 +342,23 @@ const useGroupInfo = () => {
     }
   };
 
+  const handleOpenProfile = async (item) => {
+    setOpenModal(false);
+    setTimeout(() => {
+      if (profile?.myProfile?.user_id === item?.user_id) {
+        return null;
+      }
+
+      return navigation.push('OtherProfile', {
+        data: {
+          user_id: profile.myProfile.user_id,
+          other_id: item?.user_id || item?.userId,
+          username: item?.user?.name || item?.user?.username || item.username
+        }
+      });
+    }, 100);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `setTimeout` here could lead to potential issues if the component unmounts before the timeout function is executed. This can result in a memory leak and unexpected behavior. Consider using a cleanup function in a `useEffect` hook to clear the timeout when the component unmounts.

```diff
-    setTimeout(() => {
+    const timeoutId = setTimeout(() => {
       if (profile?.myProfile?.user_id === item?.user_id) {
         return null;
       }

       return navigation.push('OtherProfile', {
         data: {
           user_id: profile.myProfile.user_id,
           other_id: item?.user_id || item?.userId,
           username: item?.user?.name || item?.user?.username || item.username
         }
       });
     }, 100);
+
+    return () => clearTimeout(timeoutId);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->